### DISCUSSION TITLE
Role of the `BuildingInsideVisualStudio` property

### DISCUSSION THREAD

**Discussion by pubudu91** [dotnet/msbuild#10505] [Q&A] [answered]
I understand that this property is there to identify if the build was triggered through Visual Studio or not. But I was wondering if setting this property does anything extra/different. i.e., does setting this property influence any behaviour of MSBuild itself (not referring to use of this property in conditions of targets)?

**Comment by GangWang01** [comment]
Probably https://learn.microsoft.com/en-us/visualstudio/msbuild/build-process-overview?view=vs-2022#visual-studio-builds-vs-msbuildexe-builds can answer it.

**Reply by pubudu91** [reply]
Thanks for the response @GangWang01. But I've already seen this and I was looking for more exact details than what's there in that doc. The section you linked to describes a couple of the major changes. Is that it? I'm more interested in the exact and complete picture of what setting the above property does.

**Comment by baronfel** [comment, marked as answer]
I did a pass on this: 
* there are some logical changes inside the [Common Targets themselves](https://github.com/search?q=repo%3Adotnet%2Fmsbuild+%22BuildingInsideVisualStudio%22+AND+path%3A*.targets&type=code)
* there's a tiny bit of engine behavior: if set, this property is passed into the SdkResolver(s) that are called to locate SDKs, so it's possible that an SdkResolver may use this information to change its behavior in some way, which would cause a project to load different SDK props or targets.


In general, I wouldn't expect huge functional changes at the engine level from this property.
